src/validationinterface.h

```diff
@@ -7,6 +7,7 @@
 #define BITCOIN_VALIDATIONINTERFACE_H
 
 #include <memory>
+#include <functional>
```

Comment:
Since you're adding system includes, might as well move them to their correct place, after the user includes.